builtin/logical/transit/path_keys.go

```diff
@@ -253,6 +261,30 @@ func (b *backend) pathPolicyRead(
 					}
 				}
 				key.Name = "ed25519"
+			case keysutil.KeyType_RSA2048, keysutil.KeyType_RSA4096:
+				key.Name = "rsa-2048"
+				if p.Type == keysutil.KeyType_RSA4096 {
+					key.Name = "rsa-4096"
+				}
+
+				// Encode the RSA public key in PEM format to return over the
+				// API
+				derBytes, err := x509.MarshalPKIXPublicKey(v.RSAKey.Public())
+				if err != nil {
+					return nil, fmt.Errorf("error marshaling RSA public key: %v", err)
+				}
+				// Not making the header 'RSA PUBLIC KEY' since the format is
+				// of generic public key and not specifically of RSA. If its
+				// 'RSA PUBLIC KEY', 'openssl' complains.
```

Comment:
Just `PUBLIC KEY` is correct since openssl (and x509) use PKIX and the type is encoded into the marshaled bytes. Can remove this comment!